x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -1444,6 +1449,457 @@ public void testIntervalParam() {
+                "from idx1 | ENRICH idx2 ON ?f1 WITH ?f2 = ?f3",
+                new QueryParams(
+                    List.of(
+                        new QueryParam("f1", "f.1.*", NULL, true),
+                        new QueryParam("f2", "f.2", NULL, true),
+                        new QueryParam("f3", "f.3*", NULL, true)
+                    )
+                )
+            )
+        );
+
+        assertEquals(
+            new Enrich(
+                EMPTY,
+                relation("idx1"),
+                null,
+                new Literal(EMPTY, "idx2", KEYWORD),
+                attribute("f.1.*.f.2"),
+                null,
+                Map.of(),
+                List.of(new Alias(EMPTY, "f.3*.f.4.*", attribute("f.5.f.6*")))
+            ),
+            statement(
+                "from idx1 | ENRICH idx2 ON ?f1.?f2 WITH ?f3.?f4 = ?f5.?f6",
+                new QueryParams(
+                    List.of(
+                        new QueryParam("f1", "f.1.*", NULL, true),
+                        new QueryParam("f2", "f.2", NULL, true),
+                        new QueryParam("f3", "f.3*", NULL, true),
+                        new QueryParam("f4", "f.4.*", NULL, true),
+                        new QueryParam("f5", "f.5", NULL, true),
+                        new QueryParam("f6", "f.6*", NULL, true)
+                    )
+                )
+            )
+        );
+    }
+
+    public void testParamForIdentifierPattern() {
+        // name patterns can appear in keep and drop
+        // all patterns
+        LogicalPlan plan = statement(
+            "from test | keep ?f1, ?f2 | drop ?f3, ?f4",
+            new QueryParams(
+                List.of(
+                    new QueryParam("f1", "f*1.", NULL, false, true),
+                    new QueryParam("f2", "f.2*", NULL, false, true),
+                    new QueryParam("f3", "f3.*", NULL, false, true),
+                    new QueryParam("f4", "f.4.*", NULL, false, true)
+                )
+            )
+        );
+
+        assertThat(plan, instanceOf(Drop.class));
+        Drop drop = (Drop) plan;
+        List<? extends NamedExpression> removals = drop.removals();
+        assertEquals(removals.size(), 2);
+        assertThat(removals.get(0), instanceOf(UnresolvedNamePattern.class));
+        UnresolvedNamePattern up = (UnresolvedNamePattern) removals.get(0);
+        assertEquals(up.name(), "f3.*");
+        assertEquals(up.pattern(), "f3.*");
+        assertThat(removals.get(1), instanceOf(UnresolvedNamePattern.class));
+        up = (UnresolvedNamePattern) removals.get(1);
+        assertEquals(up.name(), "f.4.*");
+        assertEquals(up.pattern(), "f.4.*");
+        assertThat(drop.child(), instanceOf(Keep.class));
+        Keep keep = (Keep) drop.child();
+        assertEquals(keep.projections().size(), 2);
+        assertThat(keep.projections().get(0), instanceOf(UnresolvedNamePattern.class));
+        up = (UnresolvedNamePattern) keep.projections().get(0);
```

Comment:
Replace the type of repetitive lines with EsqlTestUtils.as - along an static import this becomes:
`UnresolvedNamePattern unp  = as(keep.projections().get(0),  UnresolvedNamedPattern.class)`

There are many instances in the tests where this pattern appears.